### PR TITLE
Dashboard Improvements

### PR DESCRIPTION
**Pull request by Aqualex**
Added last10Quotes function which extracts the last 10 quotes
Added table schema for Buy/Sell Pressure
Added srcQuoteUpd function which adds spread,midprice,midsize to srcquote
Added clienttradeUpd which adds volume,volatility to clienttrade
Added buySellPressure which calculates the the number of buy orders as percentage of total orders for each sym

### CODE REVIEW

**Comment by ThomasSmyth** [comment]
That description is a chore. Is this related to a specific issue?

File: code/vizSum/vizSum.q [hunk 1]
```diff
@@ -1,18 +1,18 @@
 //DEFINE schema for tables 
 clienttradeLVC:([sym:`$()]time:`timestamp$();price:`float$();size:`int$();stop:`boolean$();cond:`char$();ex:`char$();side:`$()); 						
 srcquoteLVC:([sym:`$()]time:`timestamp$();src:`$();bid:`float$();ask:`float$();bsize:`long$();asize:`long$();mode:`char$();ex:`char$());  					
-srcquoteMid:([]time:`timestamp$();sym:`$();src:`$();bid:`float$();ask:`float$();bsize:`long$();asize:`long$();mode:`char$();ex:`char$();midPrice:`float$();midSize:`float$());  
-   
+/srcquoteMid:([]time:`timestamp$();sym:`$();src:`$();bid:`float$();ask:`float$();bsize:`long$();asize:`long$();mode:`char$();ex:`char$();midPrice:`float$();midSize:`float$());
```
Comment (claire-sloan) [inline code/vizSum/vizSum.q]: uncomment

Comment (ThomasSmyth) [inline code/vizSum/vizSum.q]: :thinking:

File: code/vizSum/vizSum.q [hunk 2]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
...
+ a:exec distinct sym from clienttrade; 
+ b:`sym xkey clienttradeLVC; 
+ update ind:i from b upsert `sym xkey -100#clienttrade  
+ } 
+
+last10Quotes:{
+ a:exec distinct sym from srcquote;
+ b:`sym xkey srcquoteLVC;
+ update ind:i from b upsert `sym xkey -100#srcquote
+ /break;
```
Comment (ThomasSmyth) [inline code/vizSum/vizSum.q]: :thinking:

File: code/vizSum/vizSum.q [hunk 3]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
 
 //Check if successfully connected to tickerplant 
 notpconnected:{[]
```
Comment (claire-sloan) [inline code/vizSum/vizSum.q]: make one line function

File: code/vizSum/vizSum.q [hunk 4]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
...
-  0 = count select from .sub.SUBSCRIPTIONS where proctype in .summary.tickerplanttypes, active
+  0=count select 
+          from .sub.SUBSCRIPTIONS where proctype in .summary.tickerplanttypes, active
```
Comment (ThomasSmyth) [inline code/vizSum/vizSum.q]: But why? :sob:

File: code/vizSum/vizSum.q [hunk 5]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
...
 
 .servers.CONNECTIONS:distinct .servers.CONNECTIONS,.summary.hdbtypes,.summary.tickerplanttypes;
 
-upd:.summary.upd;  
+upd:.summary.upd; 
+
+last10Trades:{
+ a:exec distinct sym from clienttrade; 
+ b:`sym xkey clienttradeLVC; 
+ update ind:i from b upsert `sym xkey -100#clienttrade
```
Comment (claire-sloan) [inline code/vizSum/vizSum.q]: is this meant to be -10#  for last 10 trades? same for quotes

File: code/vizSum/vizSum.q [hunk 6]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
...
+ /select 
+ /from a where 0=mod[i;10]  
+ }
+
+buySellPressure:{
+ `BSPressure upsert`sym xkey update pressure:100*press%ind 
+                             from update press:sums press,ind:sums ind 
+                                  by sym 
+                                  from update press:{$[`BUY~x;1;-1]}'[side],ind:1 
+                                       from clienttrade;BSPressure
```
Comment (ThomasSmyth) [inline code/vizSum/vizSum.q]: Excessive indentation is becoming a calling card for you.

File: code/vizSum/vizSum.q (hunk 2 above)
Comment (claire-sloan) [inline code/vizSum/vizSum.q]: remove break

File: code/vizSum/vizSum.q [hunk 7]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
...
+ /from a where 0=mod[i;10]  
+ }
+
+buySellPressure:{
+ `BSPressure upsert`sym xkey update pressure:100*press%ind 
+                             from update press:sums press,ind:sums ind 
+                                  by sym 
+                                  from update press:{$[`BUY~x;1;-1]}'[side],ind:1 
+                                       from clienttrade;BSPressure 
+ }
```
Comment (ThomasSmyth) [inline code/vizSum/vizSum.q]: semi colon.

File: code/vizSum/vizSum.q [hunk 8]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
...
+ update ind:i from b upsert `sym xkey -100#srcquote
+ /break; 
+ }
+
+srcQuoteUpd:{[st;et]
+ if[prd 00:00~'(st;et);st:09:00;et:.z.P];
+ a:update spread:ask-bid,midprice:%[bid+ask;2],midsize:%[bsize+asize;2] 
+   from srcquote where time within(st;et); 
+ 
+ select
```
Comment (claire-sloan) [inline code/vizSum/vizSum.q]: make one line select statement

File: code/vizSum/vizSum.q [hunk 9]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
...
+                             from update press:sums press,ind:sums ind 
+                                  by sym 
+                                  from update press:{$[`BUY~x;1;-1]}'[side],ind:1 
+                                       from clienttrade;BSPressure 
+ }
 
 HDBC:{
-  hdbhand:exec first w from .sub.getsubscriptionhandles[.summary.hdbtypes;();()!()];												//Get handle for hdb 
+  hdbhand:exec first w 
+          from .sub.getsubscriptionhandles[.summary.hdbtypes;();()!()];												//Get handle for hdb
```
Comment (ThomasSmyth) [inline code/vizSum/vizSum.q]: No need for indentation

File: code/vizSum/vizSum.q [hunk 10]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
...
+ select 
+ from a where 0=mod[i;10] 
+ }
+
+clienttradeUpd:{[st;et]
+ if[prd 00:00~'(st;et);st:09:00;et:.z.P-1]; 
+ a:update volume:price*size,volatility:0^{100*(log y%x)xexp 2}':[price] 
+   by sym 
+   from clienttrade where time within(st;et);a
+ /select
```
Comment (claire-sloan) [inline code/vizSum/vizSum.q]: uncomment and make one line

File: code/vizSum/vizSum.q [hunk 11]
```diff
@@ -39,18 +39,63 @@ subscribe:{[]
...
+ `BSPressure upsert`sym xkey update pressure:100*press%ind 
+                             from update press:sums press,ind:sums ind 
+                                  by sym 
+                                  from update press:{$[`BUY~x;1;-1]}'[side],ind:1 
+                                       from clienttrade;BSPressure 
+ }
 
 HDBC:{
-  hdbhand:exec first w from .sub.getsubscriptionhandles[.summary.hdbtypes;();()!()];												//Get handle for hdb 
+  hdbhand:exec first w
```
Comment (claire-sloan) [inline code/vizSum/vizSum.q]: make one line

File: code/vizSum/vizSum.q [hunk 12]
```diff
@@ -64,4 +109,4 @@ while[.summary.notpconnected[];
   .summary.subscribe[];
   ]
 
-HDBC[];
+/HDBC[];
```
Comment (claire-sloan) [inline code/vizSum/vizSum.q]: uncomment

File: code/vizSum/vizSum.q (hunk 5 above)
Comment (ThomasSmyth) [reply code/vizSum/vizSum.q]: If there less than 100 then some trades will get repeated.

File: code/vizSum/vizSum.q (hunk 6 above)
Comment (Aqualex) [reply code/vizSum/vizSum.q]: in my defence I ask Jordan if it looks good this way and he said yes. MD questions also have something similar. Will sort it out.

File: code/vizSum/vizSum.q (hunk 4 above)
Comment (Aqualex) [reply code/vizSum/vizSum.q]: 😆

File: code/vizSum/vizSum.q (hunk 6 above)
Comment (claire-sloan) [reply code/vizSum/vizSum.q]: :open_mouth:

**Pull request merged**